pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/text/LuceneTextIndexCreator.java

```diff
@@ -90,14 +102,17 @@ public static HashSet<String> getDefaultEnglishStopWordsSet() {
    *               to offline), we close this lucene index writer to release resources but don't commit.
    * @param config the text index config
    */
-  public LuceneTextIndexCreator(String column, File segmentIndexDir, boolean commit, TextIndexConfig config) {
+  public LuceneTextIndexCreator(String column, File segmentIndexDir, boolean commit, boolean realtimeConversion,
+      @Nullable int[] immutableToMutableIdMap, TextIndexConfig config) {
     _textColumn = column;
+
+    // to reuse the mutable index, it must be (1) not the realtime index, and (2) is realtime segment conversion
```

Comment:
what does (1) mean exactly? "not the realtime index"? the code says commit?

Reply:
clarified in the code - `LuceneTextIndexCreator` is used for both in `RealtimeLuceneTextIndex` and separately for the offline lucene text index. We use the `commit` flag (set only for offline index) to avoid trying to convert an index if the instance is created by `RealtimeLuceneTextIndex`